### DISCUSSION TITLE
Feature Request: Show / hide fields via checkboxes on index

### DISCUSSION THREAD

**Discussion by FY0u11** [laravel/nova-issues#5913] [Ideas & Feature Requests] [answered]
Hello there, the feature request is a copy of https://github.com/laravel/nova-issues/issues/2622. Please see it for the reference.

I tried this solution to achieve this: https://twitter.com/jbrooksuk/status/1283012864282972160

But firstly, now there is NovaRequest class, that seems not having filters() method. Secondly, even if I access to the filters via something like this code: 

```php
$filters = [];
if ($request->users_filter) {
    try {
        $filters = json_decode(base64_decode($request->users_filter), true)[0][FieldsFilter::class];
    } catch (\Throwable $exception) {}
}
```

the fieldsForIndex method itself seems to be called 52 times, and filters there only on the first call:

![image](https://github.com/laravel/nova-issues/assets/48677295/24b2fe8b-bff4-46c8-970b-0db4c2f150fb)

This is just one reloading of the resource page. As you can see, filters there only once and with this approach the https://twitter.com/jbrooksuk/status/1283012864282972160 solution won't work.

**Comment by FY0u11** [comment]
Any solutions here?

**Comment by milewski** [comment, marked as answer]
I made a package for this exact purpose: https://github.com/dcasia/column-toggler (without using filters)

**Reply by FY0u11** [reply]
Thank you
